.github/workflows/ci_pr.yml

```diff
@@ -0,0 +1,140 @@
+name: CI Unit tests
+
+on:
+  push:
+    branches: [ "*" ]
+  pull_request:
+    branches: [ "*" ]
+
+jobs:
+  legacy:
+    
+    name: "Python 2.6 Unit Tests"
+    runs-on: ubuntu-16.04
+
+    strategy:
+      fail-fast: false
+    env:
+      NOSEOPTS: "--verbose"
+
+    steps:
+
+    - name: Install Python 2.6
+      run: |
+        sudo add-apt-repository ppa:deadsnakes/ppa -y
+        sudo apt-get update -y
+        sudo apt install python2.6 python2.6-dev -y
+        sudo ln -sf /usr/bin/python2.6 /usr/bin/python
+        PATH="/usr/bin/${PATH:+:${PATH}}"
+        echo $PATH
+        echo "Installed python2.6"
+        python -V
+        curl https://bootstrap.pypa.io/2.6/get-pip.py -o get-pip.py
+        sudo env "PATH=$PATH" python get-pip.py
+
+    - uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: |
+        sudo env "PATH=$PATH" pip install -r requirements.txt
+        sudo env "PATH=$PATH" pip install -r test-requirements.txt
+    
+    - name: Test with nosetests
+      run: |
+        ./ci/nosetests.sh || EXIT_CODE=$?
```

Comment:
You havent defined exit code before, and if `./ci/nosetests.sh `  succeeds then `EXIT_CODE=$?` this command wont be run. 

For when you finally exit `exit $EXIT_CODE`, would this automatically set to `0` even if nosetest command succeeds?

Reply:
My understanding is that `$EXIT_CODE` is replaced with the empty string, and that `exit` with no arguments does in fact exit with a code of 0. However, I think it would be better to explicitly set `EXIT_CODE` before hand, if only for the sake of readability.

Reply:
Yes, or simply
`./ci/nosetests.sh; EXIT_CODE=$?`

Reply:
I think we should explicitly set `EXIT_CODE` to nosetest's exit code instead of relying on exit with no arguments defaulting to 0.

Reply:
Fixed! Resolving this thread now.